inductiva/tasks/task.py

```diff
@@ -175,11 +176,35 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
     def kill(self) -> None:
         """Kill the task.
 
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
+        This method issues a request to the API to kill the task.
+        Then waits for 5s (checking every second) to see if the task was killed.
+        If the task is still running (or has a status different from killed)
+        after 5s, it issues a warning.
         """
         self._api.kill_task(path_params=self._get_path_params())
 
+        task_status = self.get_status()
+
+        start_time = time.time()
+
+        while task_status == models.TaskStatusCode.PENDINGKILL:
+
+            time.sleep(1)
+            task_status = self.get_status()
```

Comment:
The status here can change to Killed, and the if below can still raise a warning.